pywbem_mock/_dmtf_schema.py

```diff
@@ -0,0 +1,392 @@
+        self._mof_zip_bn = '%s%s-MOFs.zip' % (self._dmtf_schema_ver,
+                                              schema_type)
+        self._mof_zip_url = \
+            'http://www.dmtf.org/standards/cim/cim_schema_v%s%s%s/%s' % \
+            (schema_ver[0], schema_ver[1], schema_ver[2], self._mof_zip_bn)
+        self._schema_mof_bld_name = self._dmtf_schema_ver + '.mof'
+
+        self._mof_zip_filename = os.path.join(self.schema_dir, self._mof_zip_bn)
+        self._schema_mof_filename = os.path.join(self.schema_mof_dir,
+                                                 self._schema_mof_bld_name)
+
+        self.verbose = verbose
+
+        # setup the schema
+        self._setup_dmtf_schema()
+
+    @property
+    def schema_ver(self):
+        """
+        :func:`py:tuple` of 3 integers with major version, minor version, and
+        revision of the DMTF Schema installed
+        """
+        return self._schema_ver
+
+    @property
+    def schema_ver_str(self):
+        """
+        :term:`string` with the DMTF schema version in the form
+        <major version>.<minor version>.<revison>.
+        """
+        return '%s.%s.%s' % (self.schema_ver[0], self.schema_ver[1],
+                             self.schema_ver[2])
+
+    @property
+    def schema_dir(self):
+        """
+        :term:`string` defining the directory in which the DMTF CIM Schema
+        mof is downloaded and expanded.
+        """
+        return self._schema_dir
+
+    @property
+    def schema_mof_dir(self):
+        """
+        :term:`string` defining the directory in which the DMTF CIM Schema
+        mof is defined. This includes the expansion of the individual schema
+        mof files. This property is very useful since it can be used as
+        the search path for compiling classes in the DMTF schema.
+        """
+        return self._schema_mof_dir
+
+    @property
+    def schema_mof_filename(self):
+        """
+        :term:`string` defining the the path for the DMTF schema
+        mof top level file name which includes the pragmas that define
+        all of the files in the DMTF schema. This filename is of the
+        general form
+        """
+        return self._schema_mof_filename
+
+    def __str__(self):
+        """
+        Return a short string representation of the
+        :class:`~pywbem_mock.DMTFSchema` object for human consumption.
+        """
+
+        return '%s(schema_ver=%s, schema_dir=%s, url=%s)' % \
+               (self.__class__.__name__, self.schema_ver_str,
+                self.schema_dir, self._mof_zip_url)
+
+    def __repr__(self):
+        """
+        Return a string representation of the :class:`~pywbem_mock.DMTFSchema`
+        object that is suitable for debugging.
+
+        The properties and qualifiers will be ordered by their names in the
+        result.
+        """
+
+        return '{0}(schema_ver={1}, schema_dir={2})'.format(
+            self.__class__.__name__, self.schema_ver, self.schema_dir)
```

Comment:
This is less than in `__str__()`. I suggest to expand this to all properties.

Reply:
I figured you would catch that one.  Added self.schema_mof_dir, self.schema_mof_filename

DISCUSSION: Should the url be part of this one since it is useful to debugging but really internal.???